src/Tabs.js

```diff
@@ -0,0 +1,233 @@
+			if ((!this.tabs[i].disabled || !core.isDef(this.tabs[i].disabled)) && !this.disabled) {
+				index = i;
+
+				break;
+			}
+		}
+
+		return index;
+	}
+
+	/**
+	 * `onClick` handler for tab items.
+	 * @param {EventFacade} event
+	 */
+	onClickItem(event) {
+		let item = event.delegateTarget;
+
+		event.preventDefault();
+
+		var index = parseInt(item.getAttribute('data-index'));
+
+		this.dispatchRequest_({
+			tab: index
+		});
+	}
+
+	/**
+	 * Removes the tab at the given index from the tabs array.
+	 * @param  {number} index
+	 * @return {Object} Returns the removed tab.
+	 */
+	removeTab(index) {
+		if (core.isNumber(index) && index > -1 && index < this.tabs.length) {
+			return this.tabs.splice(index, 1);
+		}
+	}
+
+	/**
+	 * Set the new tabs state. The state is a payload object
+	 * containing the tab number, e.g. `{tab:1}`.
+	 * @param {Object} state
+	 * @protected
+	 */
+	setState_(state) {
+		this.tab = state.tab;
+
+		this.lastState_ = state;
+	}
+
+	/**
+	 * Disables the tab at the given index in the tabs array.
+	 * @param  {number} index
+	 * @return {boolean} disabled
+	 */
+	setTabDisabled(index, disabled) {
+		if (core.isNumber(index) && core.isBoolean(disabled)) {
+			this.tabs[index].disabled = disabled;
+
+			this.tabs = this.tabs;
+		}
+	}
+
+	/**
+	 * Toggles the disabled state of the tab at the given index in the tabs array.
+	 * If the tab at the given index is active but toggled disabled the first
+	 * the first not disabled tab will be selected.
+	 * @param  {number} index
+	 */
+	toggleTabDisabled(index) {
+		if (core.isNumber(index) && index >= 0 && index < this.tabs.length) {
+			this.tabs[index].disabled = !this.tabs[index].disabled;
+
+			if (index === this.tab) {
+				this.tab = this.findFirstAvailableIndex_();
+			}
+
+			this.tabs = this.tabs;
+		}
+	}
+}
+Soy.register(Tabs, templates);
+
+Tabs.Events = {
+	CHANGE_REQUEST: 'changeRequest'
+};
+
+/**
+ * Default types used to style the tabs component.
+ */
+Tabs.TYPES = {
+	NONE: null,
+	PILLS: 'pills',
+	TABS: 'tabs'
+};
+
+/**
+ * State definition.
+ * @type {!Object}
+ * @static
+ */
+Tabs.STATE = {
+	/**
+	 * Determines if the whole component should be disabled or not.
+	 * @type {boolean}
+	 * @default false
+	 */
+	disabled: {
+		validator: core.isBoolean,
+		value: false
+	},
+
+	/**
+	 * Tab to display on initial paint. It has two attributes, 'label', which is
+	 * required and 'disabled' which is optional.
+	 * @type {number}
+	 * @default Default will be set by the valueFn.
+	 */
+	tab: {
+		validator: core.isNumber,
+		valueFn: 'findFirstAvailableIndex_'
+	},
+
+	/**
+	 * Tabs array, holding the actual tabs.
+	 * @type {Array}
+	 * @default []
+	 */
+	tabs: {
+		validator: value => value.every(item => !!item.label),
+		value: []
+	},
+
+	/**
+	 * Type currently being used to style the tabs component.
+	 * @type {string}
+	 * @default {string}
+	 */
+	type: {
+		validator: value => value.toUpperCase() in Tabs.TYPES,
```

Comment:
It seems like this wouldn't allow setting type to `Tabs.TYPES.NONE`, since its value is `null`. Maybe set `Tabs.TYPES.NONE` to `'none'` instead to keep the pattern? :)